python/lsst/ap/association/diaPipe.py

```diff
@@ -469,21 +506,22 @@ def createNewDiaObjects(self, diaSources):
               (`pandas.DataFrame`)
             - ``newDiaObjects`` : Newly created DiaObjects from the
               unassociated DiaSources. (`pandas.DataFrame`)
+            - ``nNewDiaObjects`` : Number of newly created diaObjects.(`int`)
         """
         newDiaObjectsList = []
-        for idx, diaSource in diaSources.iterrows():
-            if diaSource["diaObjectId"] == 0:
-                newDiaObjectsList.append(
-                    self._initialize_dia_object(diaSource["diaSourceId"]))
-                diaSources.loc[idx, "diaObjectId"] = diaSource["diaSourceId"]
+        for idx, diaSource in unAssocDiaSources.iterrows():
+            newDiaObjectsList.append(
+                self._initialize_dia_object(diaSource["diaSourceId"]))
```

Comment:
I would investigate if you can run this without the `iterrows` loop, for example:

`newDiaObjectsSeries = unAssocDiaSources['diaSourceId'].apply(self._initialize_dia_object)`

which if it works will likely be faster than the explicit loop.

Reply:
I'll give it a shot.

Reply:
Seems to work. I remember apply being only slightly faster than iterrows when not using built in pandas operations. Still, it's likely to be a bit faster since it now operates on one column not the whole row.